src/lib.rs

```diff
@@ -116,7 +116,7 @@ pub async fn start_lemmy_server() -> Result<(), LemmyError> {
 
   let client = ClientBuilder::new(reqwest_client.clone())
     .with(TracingMiddleware::default())
-    .with(RetryTransientMiddleware::new_with_policy(retry_policy))
+    //.with(RetryTransientMiddleware::new_with_policy(retry_policy))
```

Comment:
I'm tailing the logs rn, but its been 2 hours with no memory build up, so it looks like this might be the culprit.

Reply:
Nevermind, it crashed with a bunch of these:


lemmy_1     | 2023-02-23T19:58:39.691257Z  INFO HTTP request{http.method=GET http.scheme="http" http.host=lemmy.ml http.target=/api/v3/ws otel.kind="server" request_id=5ca57764-1f9b-4f0a-8ad9-01c268d24150}: lemmy_server::api_routes_websocket: 89.36.78.186 joined                                                                                                                       lemmy_1     | 2023-02-23T19:58:40.993832Z  INFO HTTP request{http.method=GET http.scheme="http" http.host=lemmy.ml http.target=/api/v3/ws otel.kind="server" request_id=233c482c-07a6-45f0-85f8-fe32d853830e}: lemmy_server::api_routes_websocket: 2001:861:51cc:2290:79b3:6143:ca34:9bc6 joined                                                                                             lemmy_1     | 2023-02-23T19:58:41.182759Z  WARN Error encountered while processing the incoming HTTP request: lemmy_server::root_span_builder: Timed out in bb8                                                                                               lemmy_1     |    0: lemmy_apub::activities::voting::vote_post                                                                  lemmy_1     |              at crates/apub/src/activities/voting/mod.rs:132                                                     lemmy_1     |    1: lemmy_apub::activities::voting::vote::receive                                                              lemmy_1     |              at crates/apub/src/activities/voting/vote.rs:72                                                     lemmy_1     |    2: lemmy_apub::activities::community::announce::receive                                                       lemmy_1     |              at crates/apub/src/activities/community/announce.rs:149                                             lemmy_1     |    3: lemmy_server::root_span_builder::HTTP request                                                              lemmy_1     |            with http.method=POST http.scheme="http" http.host=lemmy.ml http.target=/inbox otel.kind="server" request_id=e751a065-f26c-412b-b435-cc9e179335f0 http.status_code=400 otel.status_code="OK"                                         lemmy_1     |              at src/root_span_builder.rs:16                                                                      lemmy_1     | LemmyError { message: None, inner: Timed out in bb8, context: "SpanTrace" }

A ton of websocket joins, and then the bb8 pool times out and can't seem to recover.

Reply:
Also getting a lot of these:

lemmy_1     | 2023-02-23T20:52:15.571380Z  WARN HTTP request{http.method=POST http.scheme="http" http.host=lemmy.ml http.target=/inbox otel.kind="server" request_id=a02c6845-7a1f-48ab-9d38-6d81e44e13bb}:receive:send:send_lemmy_activity: activitypub_federation::core::activity_queue: Maximum number of activitypub workers reached. Consider increasing worker count to avoid federation delays

Reply:
I also got this one, along with some more `couldnt_get_posts: Timed out in bb8` errors;

thread '<unnamed>' panicked at 'Error connecting to postgres://lemmy:TTT@postgres:5432/lemmy', src/scheduled_tasks.rs:24:27

This tells me its definitely an issue with the connection pool being overloaded on high request volume, crashing, and not being able to recover.